src/macros.rs

```diff
@@ -7,7 +7,9 @@ macro_rules! redis_command {
      $firstkey:expr,
      $lastkey:expr,
      $keystep:expr,
-     $acl_categories:expr) => {{
+     $mandatory_acl_categories:expr
+     $(, $optional_acl_categories:expr)?
+    ) => {{
```

Comment:
For the simplicity, I'd do two things:

1. Add a separate enum: `CommandAclCategory`:
```rust
enum CommandAclCategory {
    Write,
    Read,
    Fast,
    Other(String),
}

impl std::fmt::Display for CommandAclCategory {
    // ...
}
```

  Where we would add all the already existing, commonly used acl categories which have already existed for a long time, and would always specifying a new one by having the `Other` variant.

2. If the mandatory ACL category is mandatory, why not simply omit it and add it automatically? We are not in C, so we can have two macros, with it and without. In the "without" macro branch, I'd invoke the "with" branch with the mandatory ACL category name as the module name (this is the case for all the modules we are currently adding the ACLs anyway).
3. Rename the `acl_categories` arguments, as we specify them separately, not all within one argument: `mandatory_acl_category` and `optional_acl_category`.

Reply:
1. doable. will take under advisement.
2. there is no one singular mandatory ACL category. the mandatory ACL categories are those that are mandatory for the module to work at all, and loading the module should fail if they are not added. optional ACL categories are those that we warn on failure to add, but do not fail.
3. we do specify multiple categories within one argument. as a space delimited string.

Reply:
Thanks for the response!

How we want to specify them for redis - this is the implementation detail. The interface here is in Rust, and we should allow the safest, most idiomatic way possible, which is an explicit set of type-safe values. Do you agree? For the user of this macro it shouldn't matter how exactly those end up being sent to the server, the user shouldn't care about this, nor should he know about this, if this is hidden behind the interface. This is not why this crate is simply forwarding the rust types to the C-api, right?

Reply:
Yes, okay, that I am not discussing anymore. I am talking now about the type safety and the explicit lists.